tests/integration/nm/bond_test.py

```diff
@@ -71,3 +93,116 @@ def test_bond_balance_slb():
+                Interface.IPV4: {
+                    InterfaceIPv4.ENABLED: True,
+                    InterfaceIPv4.DHCP: False,
+                    InterfaceIPv4.ADDRESS: [
+                        {
+                            InterfaceIPv4.ADDRESS_IP: IPV4_ADDRESS1,
+                            InterfaceIPv4.ADDRESS_PREFIX_LENGTH: 24,
+                        }
+                    ],
+                },
+                Interface.IPV6: {
+                    InterfaceIPv6.ENABLED: True,
+                    InterfaceIPv6.DHCP: False,
+                    InterfaceIPv6.AUTOCONF: False,
+                    InterfaceIPv6.ADDRESS: [
+                        {
+                            InterfaceIPv6.ADDRESS_IP: IPV6_ADDRESS1,
+                            InterfaceIPv6.ADDRESS_PREFIX_LENGTH: 64,
+                        }
+                    ],
+                },
+            }
+            libnmstate.apply(
+                {
+                    Interface.KEY: [
+                        {
+                            Interface.NAME: BOND0,
+                            Interface.TYPE: InterfaceType.BOND,
+                            Interface.STATE: InterfaceState.UP,
+                        },
+                        vlan_iface,
+                    ]
+                }
+            )
+            # Apply the simple configure again is the key reproducer
+            libnmstate.apply(
+                {
+                    Interface.KEY: [
+                        {
+                            Interface.NAME: BOND0,
+                            Interface.TYPE: InterfaceType.BOND,
+                        },
+                        vlan_iface,
+                    ]
+                }
+            )
+            assert retry_till_true_or_timeout(
+                RETRY_TIMEOUT, vlan_is_up_with_ip
+            )
+            cmdlib.exec_cmd("ip link set eth1 down".split(), check=True)
+            cmdlib.exec_cmd("ip link set eth2 down".split(), check=True)
+            assert retry_till_true_or_timeout(
+                RETRY_TIMEOUT, vlan_is_down_without_ip
+            )
+            yield
+
+
+def vlan_is_down_without_ip():
+    current_state = statelib.show_only((TEST_VLAN,))
+    print(f"Current state {current_state}")
+    iface_state = current_state[Interface.KEY][0]
+    return (
+        iface_state[Interface.IPV4][InterfaceIPv4.ENABLED] is False
+        and iface_state[Interface.IPV6][InterfaceIPv6.ENABLED] is False
+    )
+
+
+def vlan_is_up_with_ip():
+    current_state = statelib.show_only((TEST_VLAN,))
+    print(f"Current state {current_state}")
+    iface_state = current_state[Interface.KEY][0]
+    return (
+        iface_state[Interface.STATE] == InterfaceState.UP
+        and len(iface_state[Interface.IPV4].get(InterfaceIPv4.ADDRESS, []))
+        == 1
+        # desired IPv6 address plus link local address
+        and len(iface_state[Interface.IPV6].get(InterfaceIPv6.ADDRESS, []))
+        == 2
+    )
```

Comment:
Instead of just matching the address length, is it possible to verify VLAN of bond got reactivated with **original desired**  IPv4 address ?

Reply:
Sure, let me do that